pkg/apis/application/v1alpha1/types.go

```diff
@@ -3130,6 +3133,16 @@ func (w *SyncWindow) Validate() error {
 	return nil
 }
 
+func (w *SyncWindow) Hash() uint64 {
+	var windowBuffer bytes.Buffer
+	enc := gob.NewEncoder(&windowBuffer)
+	err := enc.Encode(w)
```

Comment:
Should it really be all fields? I am not sure `ManualSync` and `Description` should be part of what decide if a window is unique or not.